diff --git a/src/cmake/DawnCompilerChecks.cmake b/src/cmake/DawnCompilerChecks.cmake
--- a/src/cmake/DawnCompilerChecks.cmake
+++ b/src/cmake/DawnCompilerChecks.cmake
@@ -76,4 +76,7 @@ if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.28
   ]] DAWN_SUPPORTS_CXX_MODULES)
 else()
   set(DAWN_SUPPORTS_CXX_MODULES False)
 endif()
+if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+  set(DAWN_SUPPORTS_CXX_MODULES False)
+endif()
